libnmstate/nm/device.py

```diff
@@ -126,6 +126,9 @@ def _safe_activate_async(dev, connection_id):
     connection = None
     if connection_id:
         connection = client.get_connection_by_id(connection_id)
+    if connection:
```

Comment:
Should there be a check that ensures that it is only called with dev or connection_id? Is it going to be called with both but the connection_id invalid?